### DISCUSSION TITLE
Send :command to qutebrowser without switching focus

### DISCUSSION THREAD

**Discussion by sucrecacao** [qutebrowser/qutebrowser#6212] [Q&A] [answered]
Hello,

I'm using qutebrowser with my text editor Kakoune to render live preview of markdown file. 

I send command to qutebrowser: `:reload` and `:scroll-px`

I have two questions:

 1. Is there a way to create a specific session and  send the command to this specific session?
 2. Can I send those command from my cli without having the focus switching to the qutebrowser window?

**Comment by The-Compiler** [comment]
If you use a different `--basedir`, you can have a separate instance (with its own data, settings, etc.), and then also send commands to that.

You can also set `new_instance_open_target` to `tab-silent` so that it doesn't take focus - ideally that'd be the default for commands, and I think there's an issue open for that, but I can't find it...

**Reply by sucrecacao** [reply]
Thanks for the quick answer.

I have used the following:
```
qutebrowser --basedir .  -s new_instance_open_target tab-silent  :reload
```
But it still switch focus to qutebrowser. Am I doing something wrong?

**Reply by The-Compiler** [reply]
That looks like it should work. What window manager are you using? Can you start with `--debug` and show the full logs (ideally from both the running and the connecting instance)?

**Reply by sucrecacao** [reply]
I'm using i3 on lubuntu

**Reply by sucrecacao** [reply]
Here is the debug:
```
$ qutebrowser --basedir .  -s new_instance_open_target tab-silent  :reload --debug
13:21:12 DEBUG    init       earlyinit:init_log:266 Log initialized.
13:21:12 DEBUG    init       app:run:97 Initializing directories...
13:21:12 DEBUG    init       standarddir:init:319 Base directory: .
13:21:12 DEBUG    init       app:run:101 Initializing config...
13:21:12 DEBUG    config     config:set_str:464 Setting new_instance_open_target (type String) to 'tab-silent' (converted from 'tab-silent')
13:21:12 DEBUG    config     config:_set_value:323 Config option changed: new_instance_open_target = tab-silent
13:21:12 DEBUG    init       app:__init__:818 Qt arguments: ['/usr/bin/qutebrowser', '--reduced-referrer-granularity'], based on Namespace(backend=None, basedir='.', color=True, command=[':reload'], debug=True, debug_flags=[], enable_webengine_inspector=False, force_color=False, json_args=None, json_logging=False, logfilter=None, loglevel='info', loglines=2000, no_err_windows=False, nowindow=False, override_restore=False, qt_arg=None, qt_flag=None, session=None, target=None, temp_basedir=False, temp_basedir_restarted=None, temp_settings=[['new_instance_open_target', 'tab-silent']], url=[], version=False)
13:21:12 DEBUG    init       app:__init__:821 Initializing application...
13:21:12 DEBUG    ipc        ipc:send_to_running_instance:433 Connecting to /tmp/qb/runtime/ipc-973b5842e92e103fa419a43ed59d95ac
13:21:12 INFO     ipc        ipc:send_to_running_instance:438 Opening in existing instance
13:21:12 DEBUG    ipc        ipc:send_to_running_instance:450 Writing: b'{"args": [":reload"], "target_arg": null, "version": "1.6.0", "protocol_version": 1, "cwd": "/tmp/qb"}\n'
```

**Reply by The-Compiler** [reply]
Ah, I think I see where the issue is: `-s new_instance_open_target tab-silent` only sets the setting for the new instance (which communicates with the existing one and then quits), but the existing instance won't know about it (and still has the old setting).

Can you try with `--target tab-silent` instead?

**Reply by sucrecacao** [reply]
doesn't help to run ` qutebrowser --basedir .  --target tab-silent  :reload`

**Comment by Aghiles-P** [comment, marked as answer]
Any solution yo this issue ?

**Reply by The-Compiler** [reply]
Not that I'm aware of. I heard people say that `--no-window` apparently works, but I haven't tried (and it's a bit of a hack). In any case, this is tracked in #5094.
